c2corg_api/views/forum.py

```diff
@@ -24,3 +30,74 @@ def get(self):
             client.discourse_public_url, d_username)
 
         return {link: link, count: count}
+
+
+class SchemaTopicCreate(colander.MappingSchema):
+    document_id = colander.SchemaNode(colander.Int())
+    lang = colander.SchemaNode(colander.String())
+
+schema_topic_create = SchemaTopicCreate()
+
+
+def validate_topic_create(request):
+    document_id = request.validated['document_id']
+    lang = request.validated['lang']
+
+    locale = DBSession.query(DocumentLocale) \
+        .filter(DocumentLocale.document_id == document_id) \
+        .filter(DocumentLocale.lang == lang) \
+        .one_or_none()
+    if locale is None:
+        request.errors.add('body',
+                           '{}/{}'.format(document_id, lang),
+                           'Document not found')
+        return
+    request.validated['locale'] = locale
+
+    if locale.topic_id is not None:
+        request.errors.add('body',
+                           '{}_{}'.format(document_id, lang),
+                           'Topic already exists')
+
+
+# Here path is required by cornice but related routes are not implemented
+# as far as we only need collection_post to create topic in discourse
+@resource(collection_path='/forum/topics', path='/forum/topics/{id}',
+          cors_policy=cors_policy)
+class ForumTopicRest(object):
+    def __init__(self, request):
+        self.request = request
+
+    @restricted_view(schema=schema_topic_create,
+                     validators=[validate_topic_create])
+    def collection_post(self):
+        settings = self.request.registry.settings
+
+        document_id = self.request.validated['document_id']
+        lang = self.request.validated['lang']
+        locale = self.request.validated['locale']
+
+        title = "{}_{}".format(document_id, lang)
+        content = '<a href="{}">{}</a>'.format(
+                self.request.referer,
+                locale.title)
+        category = settings['discourse.category']
+        # category could be id or name
+        try:
```

Comment:
I've allowed to set category name or id in config. This could be useful for development phase, as far as category id could be different in the different databases.
In case we pass a str to create_post, it will be threated as a name. In case category is an id, I wand to convert it to an int.
In case category is a name (str), int(category) will raise an error.